### PR TITLE
feat: implement pub/sub commands

### PR DESCRIPTION
**Pull request by Parcley**
add basic support for publish subscribe channels

still missing:
- pattern support (i.e. PSUBSCRUBE PUNSUBSCRIBE)
- reject all commands when redis client enters subscribe state

### CODE REVIEW

File: src/commands/subscribe.js [hunk 1]
```diff
@@ -1,3 +1,18 @@
+import emitMessage from '../commands-utils/emitMessage';
+
 export function subscribe(...args) {
-  return args.length;
+  args.forEach(chan => {
+    if (this.channels.listenerCount(chan) === 0) {
+      this.channels.on(chan, message => {
+        emitMessage(this, chan, message);
+      });
+      // now: client shall enter the subscribed state
```
Comment (stipsan) [inline src/commands/subscribe.js]: feel free to change these back to TODO, I haven't gotten around to tweak codeclimate yet but it I disagree with many of its default "warnings" 🙂

Comment (Parcley) [reply src/commands/subscribe.js]: in fact I am working on actually implementing these and to keep the tests (and production code useable), I am thinking to work on Issue #464

**Comment by Parcley** [comment]
The mocked redisClient will now reject commands when in subscriber mode or not connected
To make redisClient usable and testable after adding this feature, I have also submitted PR #656 and included PR #656 changes in this PR

**Comment by stipsan** [comment]
ok cool! I'll review #656 first then and revisit this one when it's merged 👍 Good work! 😄

**Pull request merged**